tests/testthat/test-model-averaging.R

```diff
@@ -62,3 +65,301 @@ test_that("Model-averaging functions work", {
+    model_list   = models,
+    parameters   = c("mu_x_cont1", "mu_x_fac2t", "mu_x_fac3o"),
+    is_null_list = list(
+      "mu_x_cont1" = c(FALSE, FALSE, FALSE),
+      "mu_x_fac2t" = c(TRUE,  FALSE, TRUE),
+      "mu_x_fac3o" = c(TRUE,  TRUE,  FALSE)
+    ),
+    conditional = FALSE)
+  
+  # Test mix posteriors
+  mixed_posteriors <- mix_posteriors(
+    model_list   = models,
+    parameters   = c("mu_x_cont1", "mu_x_fac2t", "mu_x_fac3o"),
+    is_null_list = list(
+      "mu_x_cont1" = c(FALSE, FALSE, FALSE),
+      "mu_x_fac2t" = c(TRUE,  FALSE, TRUE),
+      "mu_x_fac3o" = c(TRUE,  TRUE,  FALSE)
+    ),
+    seed = 1, n_samples = 1000)
+  
+  # Checks
+  expect_true(is.list(inference))
+  expect_true(all(c("mu_x_cont1", "mu_x_fac2t", "mu_x_fac3o") %in% names(inference)))
+  expect_true(is.numeric(inference$mu_x_cont1$BF))
+  expect_true(is.numeric(inference$mu_x_fac2t$BF))
+  expect_true(is.numeric(inference$mu_x_fac3o$BF))
+  # Allow for small difference in sample size due to spike samples
+  expect_true(abs(length(mixed_posteriors$mu_x_cont1) - 1000) <= 1)
+  
+  # Visual check
+  vdiffr::expect_doppelganger("model-averaging-formulas", function(){
+    par(mfrow = c(2, 2))
+    hist(mixed_posteriors$mu_x_cont1, main = "mu_x_cont1")
+    hist(mixed_posteriors$mu_x_fac2t, main = "mu_x_fac2t")
+    if(is.matrix(mixed_posteriors$mu_x_fac3o)) {
+      hist(mixed_posteriors$mu_x_fac3o[,1], main = "mu_x_fac3o[1]")
+      hist(mixed_posteriors$mu_x_fac3o[,2], main = "mu_x_fac3o[2]")
+    }
+  })
+})
+
+
+# ==============================================================================
+# HELPER FUNCTION FOR PRINT TESTS
+# ==============================================================================
+# Generate ensemble inference for print testing
+generate_print_test_inference <- function(temp_fits_dir) {
+  # Load pre-fitted model and its marginal likelihood
+  fit_simple_normal <- readRDS(file.path(temp_fits_dir, "fit_simple_normal.RDS"))
+  marglik_normal <- readRDS(file.path(temp_fits_dir, "fit_simple_normal_marglik.RDS"))
+  
+  # Create test data and models for averaging
+  set.seed(1)
+  data <- list(
+    x = rnorm(50, 0, .5),
+    N = 50
+  )
+  
+  log_posterior_normal <- function(parameters, data){
+    sum(stats::dnorm(data$x, parameters[["m"]], parameters[["s"]], log = TRUE))
+  }
+  
+  priors_spike <- list(
+    m = prior("spike", list(0)),
+    s = prior("normal", list(0, 1), list(0, Inf))
+  )
+  
+  model_syntax <-
+    "model
+    {
+      for(i in 1:N){
+        x[i] ~ dnorm(m, pow(s, -2))
+      }
+    }"
+  
+  fit_spike <- JAGS_fit(model_syntax, data, priors_spike, 
+                        chains = 2, adapt = 100, burnin = 150, sample = 500, seed = 2)
+  
+  # Get marginal likelihood for spike model
+  marglik_spike <- JAGS_bridgesampling(fit_spike, 
+                                        log_posterior = log_posterior_normal, 
+                                        data = data, prior_list = priors_spike)
+  
```

Comment:
again, all fitting and marglik should be done in the fit file -- check the rest file as well